src/gui/src/dbDescriptors.cpp

```diff
@@ -230,6 +230,206 @@ static odb::dbTechLayer* getLayerSelection(odb::dbTech* tech,
+    tech_layers.insert(gui->makeSelected(tech_layer));
+  }
+  props.push_back({"Tech Layers", tech_layers});
+
+  SelectionSet tech_vias;
+  for (auto tech_via : tech->getVias()) {
+    tech_vias.insert(gui->makeSelected(tech_via));
+  }
+  props.push_back({"Tech Vias", tech_vias});
+
+  std::vector<odb::dbTechSameNetRule*> rule_samenets;
+  tech->getSameNetRules(rule_samenets);
+  SelectionSet samenet_rules;
+  for (auto samenet : rule_samenets) {
+    samenet_rules.insert(gui->makeSelected(samenet));
+  }
+  props.push_back({"Same-Net Rules", samenet_rules});
+
+  return props;
+}
+
+Selected DbTechDescriptor::makeSelected(std::any object) const
+{
+  if (auto tech = std::any_cast<odb::dbTech*>(&object)) {
+    return Selected(*tech, this);
+  }
+  return Selected();
+}
+
+bool DbTechDescriptor::lessThan(std::any l, std::any r) const
+{
+  auto l_tech = std::any_cast<odb::dbTech*>(l);
+  auto r_tech = std::any_cast<odb::dbTech*>(r);
+  return l_tech->getId() < r_tech->getId();
+}
+
+bool DbTechDescriptor::getAllObjects(SelectionSet& objects) const
+{
+  auto tech = db_->getTech();
+  if (tech == nullptr) {
+    return false;
+  }
+  objects.insert(makeSelected(tech));
+  return true;
+}
+
+//////////////////////////////////////////////////
+
+DbBlockDescriptor::DbBlockDescriptor(odb::dbDatabase* db) : db_(db)
+{
+}
+
+std::string DbBlockDescriptor::getName(std::any object) const
+{
+  auto block = std::any_cast<odb::dbBlock*>(object);
+  return block->getName();
+}
+
+std::string DbBlockDescriptor::getTypeName() const
+{
+  return "Block";
+}
+
+bool DbBlockDescriptor::getBBox(std::any object, odb::Rect& bbox) const
+{
+  auto block = std::any_cast<odb::dbBlock*>(object);
+  bbox = block->getBBox()->getBox();
+  return true;
+}
+
+void DbBlockDescriptor::highlight(std::any object, Painter& painter) const
+{
+  auto block = std::any_cast<odb::dbBlock*>(object);
+
+  odb::dbBox* bbox = block->getBBox();
+  odb::Rect rect = bbox->getBox();
+  painter.drawRect(rect);
+}
+
+Descriptor::Properties DbBlockDescriptor::getProperties(std::any object) const
+{
+  auto block = std::any_cast<odb::dbBlock*>(object);
+
+  auto gui = Gui::get();
+
+  Properties props;
+  SelectionSet bterms;
+  for (auto bterm : block->getBTerms()) {
+    bterms.insert(gui->makeSelected(bterm));
+  }
+  props.push_back({"BTerms", bterms});
+
+  SelectionSet nets;
+  for (auto net : block->getNets()) {
+    nets.insert(gui->makeSelected(net));
+  }
+  props.push_back({"Nets", nets});
+
+  SelectionSet regions;
+  for (auto region : block->getRegions()) {
+    regions.insert(gui->makeSelected(region));
+  }
+  props.push_back({"Regions", regions});
+
+  SelectionSet insts;
+  for (auto inst : block->getInsts()) {
+    insts.insert(gui->makeSelected(inst));
+  }
+  props.push_back({"Insts", insts});
+
+  SelectionSet iterms;
+  for (auto iterm : block->getITerms()) {
+    iterms.insert(gui->makeSelected(iterm));
+  }
+  props.push_back({"ITerms", iterms});
```

Comment:
Let's drop this.  You can access these through the dbInst and this list is likely to be huge.

Reply:
Should we keep the link to block inspection in the iterms?

Reply:
No - they should just link to the instance.  The normal hierarchy is block<->inst<->iterm.